crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -313,30 +331,41 @@ impl JwtAuthHttpProxy {
 
         Ok(event_response_body.id)
     }
-}
 
-#[async_trait]
-impl C8YHttpProxy for JwtAuthHttpProxy {
-    fn url_is_in_my_tenant_domain(&self, url: &str) -> bool {
-        self.end_point.url_is_in_my_tenant_domain(url)
+    async fn get_c8y_end_point(&mut self) -> Result<C8yEndPoint, SMCumulocityMapperError> {
+        match &self.end_point {
+            either::Right(c8y_end_point) => {
+                return Ok(c8y_end_point.clone());
+            }
+            either::Left(ref _uninitialised) => {}
+        }
+        self.init().await
     }
 
     #[instrument(skip(self), name = "init")]
-    async fn init(&mut self) -> Result<(), SMCumulocityMapperError> {
-        info!("Initialisation");
-        while self.end_point.c8y_internal_id.is_empty() {
-            if let Err(error) = self.try_get_and_set_internal_id().await {
-                error!(
-                    "An error ocurred while retrieving internal Id, operation will retry in {} seconds and mapper will reinitialise.\n Error: {:?}",
-                    RETRY_TIMEOUT_SECS, error
-                );
-
-                tokio::time::sleep(Duration::from_secs(RETRY_TIMEOUT_SECS)).await;
-                continue;
-            };
+    async fn init(&mut self) -> Result<C8yEndPoint, SMCumulocityMapperError> {
+        if self.end_point.is_left() {
+            info!("Initialisation");
+
+            let uninitialised = self.end_point.as_ref().unwrap_left().clone();
+            let c8y_end_point = match self.try_get_internal_id().await {
+                Ok(internal_id) => Ok(C8yEndPoint::new(&uninitialised, internal_id)),
+                Err(_error) => Err(SMCumulocityMapperError::FailedToRetrieveC8yInternalId),
+            }?;
+            self.end_point = either::Either::Right(c8y_end_point);
+            info!("Initialisation done.");
         }
```

Comment:
Wh ynot just go with `if let` construct here (sorry for pseudo code):
```suggestion
        if let Uninitialized = self.end_point {
            info!("Initialisation");

            let uninitialised = self.end_point.as_ref().unwrap_left().clone();
            let c8y_end_point = match self.try_get_internal_id().await {
                Ok(internal_id) => Ok(C8yEndPoint::new(&uninitialised, internal_id)),
                Err(_error) => Err(SMCumulocityMapperError::FailedToRetrieveC8yInternalId),
            }?;
            self.end_point = either::Either::Right(c8y_end_point);
            info!("Initialisation done.");
        }
```

Reply:
Indeed, this will avoid the following `unwrap_left()`.